Sources/SecureXPC/Client/XPCClient.swift

```diff
@@ -126,16 +126,36 @@ public class XPCClient {
         XPCMachClient(serviceName: machServiceName)
     }
 
-	// MARK: Implementation
+	public static func forEndpoint(_ endpoint: XPCServerEndpoint) -> XPCClient {
+        let connection = xpc_connection_create_from_endpoint(endpoint.endpoint)
 
-    internal let serviceName: String
+        xpc_connection_set_event_handler(connection, { (event: xpc_object_t) in
+            // A block *must* be set as the handler, even though this block does nothing.
+            // If it were not set, a crash would occur upon calling xpc_connection_resume.
+            xpc_connection_cancel(event)
```

Comment:
Good Q. I cargo-culted it from the WWDC sessions I linked to in #22.

I meant to come back to this and figure out why they were doing that. Got any ideas? As far as I can tell, this closure is never called.

Reply:
I very much don't, it looks rather bizarre  I was hoping you could explain why it was added.

My understanding is `xpc_connection_cancel` expects an `xpc_connection_t` and due to typealiasing its compile time valid to pass in an `xpc_object_t`. However, for a client to receive a connection would be quite unexpected. And then why would it cancel be called on it as soon as its received?

Reply:
My usage here is a mistaken representation of what they did in their sample.

The correct equivalent would be `xpc_connection_cancel(connection)`. The idea is:

1. They needed something to hold on to `connection` to prevent arc from cleaning it up. In this case, they made their handler capture it, as a handy way to keep the connection alive until its cancelled by some other means.
2. They never expected this block to be called. I think they're using `xpc_connection_cancel` as a kind of "fatal error".

Here's the transcript from the relevant part of the video, btw:

https://github.com/ASCIIwwdc/wwdc-session-transcripts/blob/2a081ef69334d043bb2475caf1a93eb4fdcd3889/2013/702.vtt#L1725-L1763

Now that I understand why they were doing this, I'll be changing my usage here.

Reply:
I've updated this to call `fatalError`. I really don't think it should ever end up being called.